### PR TITLE
Report fixes

### PR DESCRIPTION
**Pull request by ndittren**
* Added a couple of tests and changed some function names to more accurately reflect what's going on

### CODE REVIEW

File: quizcon/main/templatetags/quiz_tools.py [hunk 1]
```diff
@@ -153,15 +153,18 @@ def right_answers_per_quiz(id):
         questions.update({total_right_answers(question): question})
```
Comment (sdreher) [inline quizcon/main/templatetags/quiz_tools.py:153]: This tally approach only records a single question for a given number of right answers, e.g. if Question 1 has 2 correct answers and Question 2 has 2 correct answers, the questions dictionary will look like this after the forloop at Ln. 149 is complete.

`{2: <Question: Question object (2)>}`

The iteration of the questions dictionary at Ln 154 *is* attempting to account for multiple questions with the same score but will not succeed as the data will not be present from the first iteration.

The same issue exists for the functions `questions_most_incorrect` and `questions_most_idk`. 

For this analytics page, I wonder if a better / more interesting view would be to list each question, and then have columns for # correct, # idk & #incorrect?

File: quizcon/main/tests/test_views.py [hunk 2]
```diff
@@ -393,23 +396,124 @@ def test_no_submissions(self):
...
+
+        qres.selected_position = 12
+        qres.save()
+
+        self.assertEqual(total_right_answers(qres.question), 0)
+        self.assertEqual(total_wrong_answers(qres.question), 0)
+        self.assertEqual(total_idk_answers(qres.question), 1)
+
+    def test_questions_most(self):
+        self.submission = QuizSubmissionFactory(
```
Comment (sdreher) [inline quizcon/main/tests/test_views.py:485]: To better tease out bugs, maybe add two additional submissions here?

File: quizcon/main/templatetags/quiz_tools.py (hunk 1 above)
Comment (ndittren) [reply quizcon/main/templatetags/quiz_tools.py:153]: Ah I see it. Yes originally I was only looking for one question per section but realized that there may be more than one question that meets the same criteria. Thinking back I don't think having this section account for more than one question was really discussed (Mike's original plan was one). I like the idea of having a questions table.

File: quizcon/templates/main/quiz_report.html [hunk 3]
```diff
@@ -159,10 +163,14 @@ <h2 class="mt-4">
                             </div>
                             <div class="card-body">
                                 <p class="card-text">
-                                    {% wrong_answers_per_quiz quiz.id as question_most_wrong %}
-                                    {% for question in question_most_wrong %}
-                                        <a href="#q{{question.pk}}" class="linked-question">{{question.text|safe}}</a>
-                                    {% endfor %}
+                                    {% questions_most_incorrect quiz.id as question_most_wrong %}
+                                    {% if question_most_wrong == "No questions answered incorrectly." %}
+                                            <b>{{question_most_right}}</b>
```
Comment (sdreher) [inline quizcon/templates/main/quiz_report.html:168]: Should be "question_most_wrong"

File: quizcon/templates/main/quiz_report.html [hunk 4]
```diff
@@ -174,10 +182,14 @@ <h2 class="mt-4">
                             </div>
                             <div class="card-body">
                                 <p class="card-text">
-                                    {% idk_answers_per_quiz quiz.id as question_most_idk %}
-                                    {% for question in question_most_idk %}
-                                        <a href="#q{{question.pk}}" class="linked-question">{{question.text|safe}}</a>
-                                    {% endfor %}
+                                    {% questions_most_idk quiz.id as question_most_idk %}
+                                    {% if question_most_idk == "No questions answered 'I don't know.'" %}
+                                            <b>{{question_most_right}}</b>
```
Comment (sdreher) [inline quizcon/templates/main/quiz_report.html:187]: Should be "question_most_idk"

File: quizcon/main/templatetags/quiz_tools.py [hunk 5]
```diff
@@ -141,51 +141,43 @@ def percentage_choice(x, question):
...
+        num_of_correct.update({question: total_right_answers(question)})
+    max_num_correct = max(num_of_correct.values())
 
-        questions.update({total_right_answers(question): question})
-    for key in questions:
-        if key == num_of_correct:
-            question_most_correct.append(questions[key])
-    return question_most_correct
+    if max_num_correct == 0:
+        return "No questions answered correctly."
```
Comment (sdreher) [inline quizcon/main/templatetags/quiz_tools.py:152]: Suggestion for all three of the functions. Maybe return an empty array if the result == 0? Then, in the template, you can check the length of the array and print out the error message. This approach keeps all the messaging in the template.

File: quizcon/templates/main/quiz_report.html [hunk 6]
```diff
@@ -144,10 +144,14 @@ <h2 class="mt-4">
...
-                                    {% right_answers_per_quiz quiz.id as question_most_right %}
-                                    {% for question in question_most_right %}
-                                        <a href="#q{{question.pk}}"  class="linked-question">{{question.text|safe}}</a>
-                                    {% endfor %}
+                                    {% questions_most_correct quiz.id as question_most_right %}
+                                    {% if question_most_right == "No questions answered correctly." %}
+                                            <b>{{question_most_right}}</b>
+                                    {% else %}
+                                        {% for question in question_most_right %}
+                                            <a href="#q{{question.pk}}" class="linked-question">{{question.text|safe}}</a>
```
Comment (sdreher) [inline quizcon/templates/main/quiz_report.html:152]: Add a `<br/>` after each anchor tag.

**Pull request merged**